File: OCTOBER-2025/25-10-2025/p1716.java
class Solution {
    public int totalMoney(int n) {
        int res = 0, avg = 4;
        while(n > 7) {
            res += avg++ * 7;
            n-=7;
        }
        while(n-- > 0) {
            res += n + avg - 3;
        }
        return res;
    }
}

File: OCTOBER-2025/25-10-2025/p1716.py
class Solution:
    def totalMoney(self, n: int) -> int:
        def calculateRemainingDaysMoney(num, week, res):
            while num > 0:
                res += week
                week += 1
                num -= 1
            return res
        if n < 7:
            return calculateRemainingDaysMoney(n, 1, 0)
        fix = 28
        weeks = n // 7
        remaining = n % 7
        res = 0        
        while weeks > 0:
            res += fix
            fix += 7
            weeks -= 1
        if remaining > 0:
            k = n // 7
            return calculateRemainingDaysMoney(remaining, k+1, res)
        return res